docs/design/fabric-v2+/fpc-registration.puml

```diff
@@ -12,261 +12,155 @@
-  Note that with the **integrated** approach we have deal with a mix of applications transactions EP and FPC-lifecycle EP.
-end note
-
-ref over User1, ERCC1
-  Create channel; each peer joins channel; each peer launches TLCC, which will  "imprint" itself with CH_id (content-addressable channel identifier, hash of gensis block)
+ref over ERCC1, ERCC2
+  It is assumed that: channel has already been created; ERCC has been installed.
+  Also: ERCC contains hard-coded the TLCC_MRenclave and the IAS Root Certificate.
 end ref
-|||
 
-ref over User1, ERCC1
-  Install and approve ERCC chaincode. ERCC defines TLCC_MR and IAS_PK in chaincode definition.
-  (alternatively, TLCC_MR and IAS_PK could be in channel configuration; crucial is that they are approved by humans (admins) on sufficiently many organizations)
-  ERCC also reads CH_id from ledger or, if that does not work, also embeds it into chaincode
-
-  see fpc-lifecycle-v2.puml
-end ref
-|||
-
-
-group enclave creation and attestation generation
-  
-  activate Admin
-
-  Admin -> Manager1 ++: create (CC_NAME, CC_VER, CH_name)
-  note right
-    CC_NAME: chaincode name
-    CC_VER: chaincode version
-    CH_NAME: channel name (needed by the ECC_Manager to cc2cc ECC
-  end note
-  note over Manager1 #FFaa00
+group enclave creation
+  note over Admin #FFaa00
     **MST**: we might want to replace (here and elsewhere) CC_VER with the sequence
     number from chain code definition as versions can be re-used in downgrades yet we
     might or might not want to re-use enclaves?
   end note
 
-  Manager1 -> Manager1 : get SPID and IAS credentials
-  note right: from local peer config
-
-  Manager1 -> ECC1 ++: create (SPID)
-  note left: this is a cc2cc call
-
-
-  ECC1 -> TLCC1 ++: getInfo
-  note over TLCC1
-    This information could also be retrieved elsewhere if easier. At this point in time
-    it does not have to be authenticated (and is anyway passed through untrusted ECC),
-    consistency will be verified later in registration (Remember that TLCC_MR is defined
-    and agreed by human consensus in the ERCC chaincode definition)
-  end note
-  return TLCC_MR, CH_id
-
-
-  |||
-
-  create participant "ECC_Enclave" as Enclave1 order 30 #99FF99
-  ECC1 -> Enclave1 ++: create (SPID, CC_NAME, CC_VER, Org_1, TLCC_MR, CH_id)
-  note left
-    this is a cgo call
-    Org_1 denotes the peer's MSP_ID
-  end note
+  Admin -> ECC1 ++: invoke create(SPID)
+
+  'ECC1 -> TLCC1 ++: getInfo
+  'note over TLCC1
+  '  This information could also be retrieved elsewhere if easier. At this point in time
+  '  it does not have to be authenticated (and is anyway passed through untrusted ECC),
+  '  consistency will be verified later in registration (Remember that TLCC_MR is defined
+  '  and agreed by human consensus in the ERCC chaincode definition)
+  'end note
+  'return TLCC_MR, CH_id
+
+  ECC1 -> Enclave1 ++: invoke create(SPID)
```

Comment:
as discussed last week, i think it would be clean (as in simpler from a overall protocol perspective and better in supporting also dcap) when the spid owning and ias interaction is done peer-side.

Reply:
* a peer component (e.g., enclave manager) should take care of that; and we would require all the control APIs up to the enclave
* if we can embed these ops in the external builder scripts, then those would be peer-side
(no strong opinion on that... as long as it does not delay the implementation)